src/pages/MergePositions/Contents.tsx

```diff
@@ -164,74 +194,234 @@ export const Contents = () => {
               ConditionalTokensService.getCollectionId(
                 parentCollectionId,
-                condition.id,
+                conditionId,
                 indexSetOfMergedPosition
               )
             )
           )
         }
 
-        setStatus(Status.Ready)
+        setTransactionStatus(Remote.success(true))
       } else {
         connect()
       }
     } catch (err) {
-      setStatus(Status.Error)
-      setError(err)
+      setTransactionStatus(Remote.failure(err))
       logger.error(err)
     }
-  }, [positions, condition, statusContext, CTService, amount, connect])
+  }, [selectedPositions, position, conditionId, condition, status, CTService, amount, connect])
 
-  const clearComponent = useCallback(() => {
-    setAmount(ZERO_BN)
-    clearPositions()
-    clearCondition()
-    updateBalances([])
-    setMergeResult('')
-    setStatus(null)
-  }, [clearPositions, clearCondition, updateBalances])
-
-  const fullLoadingActionButton =
-    status === Status.Error
-      ? {
-          buttonType: ButtonType.danger,
-          onClick: () => setStatus(null),
-          text: 'Close',
+  const advancedFilters: AdvancedFilterPosition = useMemo(() => {
+    return {
+      CollateralValue: {
+        type: null,
+        value: null,
+      },
+      ToCreationDate: null,
+      FromCreationDate: null,
+      TextToSearch: {
+        type: PositionSearchOptions.All,
+        value: null,
+      },
+      WrappedCollateral: WrappedCollateralOptions.All,
+    }
+  }, [])
+
+  const { data: positions, error: errorPositions } = usePositionsList(advancedFilters)
+
+  const onRowClicked = useCallback(
+    async (position: PositionWithUserBalanceWithDecimals) => {
+      setPosition(position)
+
+      setMergeablePositions([])
+      setConditionIds([])
+
+      if (positions && positions.length > 0 && !!position) {
+        setIsLoadingConditions(true)
+        setIsLoadingMergeablePositions(true)
+
+        const positionsFiltered = positions.filter(
+          (positionWithBalance: PositionWithUserBalanceWithDecimals) =>
+            //#1 Filter, only positions with balance
+            !positionWithBalance.userBalanceERC1155.isZero() &&
+            //#2 Filter, remove original selected position
+            positionWithBalance.id.toLowerCase() !== position.id.toLowerCase() &&
+            //#3 Filter, positions with the same condition set
+            positionWithBalance.conditionIds.sort().join('') ===
+              position.conditionIds.sort().join('') &&
+            //#4 Filter, positions with the same collateral
+            positionWithBalance.collateralToken.toLowerCase() ===
+              position.collateralToken.toLowerCase() &&
+            //#5 Filter, allow only disjoint positions
+            positionWithBalance.conditions.filter((condition: ConditionInformation) =>
+              isDisjointPartition(positionWithBalance.indexSets, condition.outcomeSlotCount)
+            ).length > 0
+        )
```

Comment:
I tried with a set of positions to test this from https://gist.github.com/hpmaxi/e88f0adcb07c944d821bf862d53c0359 and found that the last predicate was false for mergeable positions. Upon further research, I think there is something wrong with that logic (but I can be completely wrong about it).

`isDisjointPartition` is used to check for splittable / mergeable positions but in our case this is passing `indexSets` from different conditions of a position to check if some of those generate a partition valid for any of the conditions. So for example a position that takes the first outcome of each split with depth > 1 will have something like [1, 1, 1]. For example, this one `0x42d702c03a40c336f563e38e63efb65b3909ec7f90dac86fb389bdc37baa46c0` has [1, 1] so there no possible condition that gives us a disjoint result.

Testing with
``` 
 0x442c735d714b0c762b322506d23eb7ce076d524af645b837cae5f622ff285c30
 0x7273b4d47ff6e588fa7f5450496d61c4ab4bd5e252e18d5329ab7a869f010aad
 0x07897cd8ae02b015b41ed45b129e307ececcc1e83228868a742127838f3c05cc
 0x35ee6129176e6c08360df5ba5206d82040ac02beb46cc469da4af85fda5777b1
```
also generates an error when they are positions of depth = 1, in other words an indexSets array of length 1, it can't be disjoint, checking the code `  if (partition.length <= 1 || outcomeSlotCount === 0) {` is false, so any position of depth 1 can't be merged with any other.

A possible alternative is to do something like this.

```javascript
 const positionsWithBalance = positions.filter(
          (position: PositionWithUserBalanceWithDecimals) =>
            //#1 Filter, only positions with balance
            !position.userBalanceERC1155.isZero()
        )

        const conditionsOfSelectedPosition = position.conditions

        const positionsFiltered = positionsWithBalance.filter((p) =>
          conditionsOfSelectedPosition.some((condition) =>
            arePositionMergeablesByCondition(
              [position, p],
              condition.conditionId,
              condition.outcomeSlotCount
            )
          )
        )
``` 

With the conditions of the selected position and filtering initially the ones without balance, we can proceed with filtering that any pair formed by the selected position and any other leveraging some of the helpers in `tools`, and in the case of `[p, p]`, this is disjoint (the only thing not explicitly considered in `arePositionMergeablesByCondition`).